mantidproject/mantid
# Change the grid button to toggle on and off

**Description of work.**
This also involves some changes in the view as it keeps a grid on-off state
This makes sure they remain synchronised.

The grid is still turned on when going non-orthogonal.

**To test:**
1. Open Workbench
1. Create an MD workspace that can be shown non-orthogonally (see script below)
1. Check that the grid button works and toggles the button on and off and the state tracks what is in the image.
1. Click the non-orthogonal axis button, the grid should be on the data, and the button appears on.
1. Check the that button still works sensibly with non-orthogonal axes.
1. Unclick non-orthogonal axes and the grid should retain whatever state it was set to.


``` python 
from mantid.simpleapi import *

SXD23767 = Load(Filename='SXD23767.raw', LoadMonitors='Exclude')
# Set some UB with angles we can play with
SetUB(SXD23767, 1,1,2,90,90,120)
mdws = ConvertToDiffractionMDWorkspace(InputWorkspace='SXD23767',
                                       OutputDimensions='HKL')
```
Fixes #28688


*This does not require release notes* because it is minor and already covered by other release notes comments about improving the slice viewer interface.

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
